### PR TITLE
Rightsizing codejail environments to t3a.micro

### PR DESCRIPTION
**Pull request by Ardiea**
## Description
Sets the default instance type to 't3a.micro' for a codejail instance. The docker container for a non production codejail instance fits nicely @ ~330MB of memory usage and essentially zero CPU usage. Honestly these maybe even could be t3a.nano but lets come back to that another time... 

Also sets the min/max/desired consistently across tiers and environments. Lots of codejail servers doing a whole lot of nothing out there at the moment. 

## Motivation and Context
Save money. 

## How Has This Been Tested?
Untested, low risk. Verified memory + CPU usage across several tiers and environments.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Enhancement (improves on existing behavior)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project. (Did you install and run the pre-commit hooks?)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Ardiea** [comment]
This one is particularly egregious. Across the mitx production environment alone this will save $123.48/month, or almost $1500/year.

File: src/ol_infrastructure/applications/codejail/Pulumi.applications.codejail.mitx.Production.yaml [hunk 1]
```diff
@@ -4,8 +4,8 @@ encryptedkey: AQICAHg/+QzF9hGIaoayDitgnEVHEhuaANONVQQOnqpkIsol1gFmLVxyy2hwwSaJxQ
 config:
   aws:region: us-east-1
   codejail:auto_scale:
-    desired: 5
-    max: 10
+    desired: 2
```
Comment (blarghmatey) [inline src/ol_infrastructure/applications/codejail/Pulumi.applications.codejail.mitx.Production.yaml:7]: We should probably keep this at 3 for a minimum. The steady state usage is low, but we periodically get large peaks when classes run an exam that require a codejail.

File: src/ol_infrastructure/applications/codejail/Pulumi.applications.codejail.mitxonline.Production.yaml [hunk 2]
```diff
@@ -4,8 +4,8 @@ encryptedkey: AQICAHg/+QzF9hGIaoayDitgnEVHEhuaANONVQQOnqpkIsol1gFMzAtzjNTziE5CH3
 config:
   aws:region: us-east-1
   codejail:auto_scale:
-    desired: 3
-    max: 10
+    desired: 2
```
Comment (blarghmatey) [inline src/ol_infrastructure/applications/codejail/Pulumi.applications.codejail.mitxonline.Production.yaml:7]: We should probably keep this at 3 for a minimum as well to ensure we have enough capacity if it's needed.

File: src/ol_infrastructure/applications/codejail/Pulumi.applications.codejail.xpro.Production.yaml [hunk 3]
```diff
@@ -4,8 +4,8 @@ encryptedkey: AQICAHg/+QzF9hGIaoayDitgnEVHEhuaANONVQQOnqpkIsol1gFIp45VyYwQaGk+qW
 config:
   aws:region: us-east-1
   codejail:auto_scale:
-    desired: 5
-    max: 10
+    desired: 2
```
Comment (blarghmatey) [inline src/ol_infrastructure/applications/codejail/Pulumi.applications.codejail.xpro.Production.yaml:7]: Same minimum of 3 here. Apparently xpro has a significant usage of codejail in their courses.

**Review by blarghmatey — APPROVED** [review]

**Pull request merged**